### ISSUE TITLE
explain statement with verbose and json format result in QD crashed 

### ISSUE THREAD

**Issue by chenbaggio** [greenplum-db/gpdb#9819]
### Greenplum version or build
gp6.0_stable
### OS version and uname -a
Linux i-cxkof350 3.10.0-862.el7.x86_64 #1 SMP Fri Apr 20 16:44:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
### Expected behavior

report query plan

### Actual behavior

QD crashed

### Step to reproduce the behavior

1.  建立表

```
create table append_rel(att1 int, att2 int);
create table append_rel1(att3 int) INHERITS (append_rel);
create table append_rel2(att4 int) INHERITS(append_rel);
```

2. 导入数据

```
insert into append_rel values(1,10),(2,20),(3,30);

```

1. 执行查询语句

```
explain (verbose, format json) with test as (select * from (select * from append_rel) p where att1 in (select att1 from append_rel where att2 >= 19) ) select att2 from append_rel where att1 in (select att1 from test where att2 <= 21);

```

**Comment by chenbaggio** [comment]
sorry! `postgres` must be compiled with debug options

**Comment by chenbaggio** [comment]
stack
```
#0  0x00007f3053f2f207 in raise () from /lib64/libc.so.6
#1  0x00007f3053f308f8 in abort () from /lib64/libc.so.6
#2  0x0000000000da5c68 in ExceptionalCondition (conditionName=0x1080f19 "!(fplan->flow)", errorType=0x107fa46 "FailedAssertion", fileName=0x1080af6 "explain.c", lineNumber=996) at assert.c:66
#3  0x00000000007ef742 in show_dispatch_info (slice=0x1c11e40, es=0x7fff338e9360, plan=0x1c05a68) at explain.c:996
#4  0x00000000007f0878 in ExplainNode (planstate=0x1c29bc8, ancestors=0x1ca5590, relationship=0x1081967 "Outer", plan_name=0x0, es=0x7fff338e9360) at explain.c:1651
#5  0x00000000007f2ab3 in ExplainNode (planstate=0x1c14890, ancestors=0x1ca5590, relationship=0x1081974 "Subquery", plan_name=0x0, es=0x7fff338e9360) at explain.c:2306
#6  0x00000000007f2d01 in ExplainNode (planstate=0x1c13f90, ancestors=0x1ca5590, relationship=0x1081967 "Outer", plan_name=0x0, es=0x7fff338e9360) at explain.c:2356
#7  0x00000000007f2ab3 in ExplainNode (planstate=0x1c139b8, ancestors=0x1ca5590, relationship=0x1081967 "Outer", plan_name=0x0, es=0x7fff338e9360) at explain.c:2306
#8  0x00000000007f2ab3 in ExplainNode (planstate=0x1c12f30, ancestors=0x1ca5590, relationship=0x108197d "SubPlan", plan_name=0x1bd39f0 "SubPlan 1", es=0x7fff338e9360) at explain.c:2306
#9  0x00000000007f442a in ExplainSubPlans (plans=0x1c7c840, ancestors=0x1ca5590, relationship=0x108197d "SubPlan", es=0x7fff338e9360, sliceTable=0x0) at explain.c:3157
#10 0x00000000007f2d43 in ExplainNode (planstate=0x1c7a6b8, ancestors=0x1ca5590, relationship=0x1082368 "Member", plan_name=0x0, es=0x7fff338e9360) at explain.c:2365
#11 0x00000000007f4371 in ExplainMemberNodes (plans=0x1c086d8, planstates=0x1c7a540, ancestors=0x1ca5590, es=0x7fff338e9360) at explain.c:3127
#12 0x00000000007f2bf5 in ExplainNode (planstate=0x1c7a0d0, ancestors=0x1ca5590, relationship=0x1081967 "Outer", plan_name=0x0, es=0x7fff338e9360) at explain.c:2331
#13 0x00000000007f2ab3 in ExplainNode (planstate=0x1c799e0, ancestors=0x1ca5590, relationship=0x0, plan_name=0x0, es=0x7fff338e9360) at explain.c:2306
#14 0x00000000007ef10f in ExplainPrintPlan (es=0x7fff338e9360, queryDesc=0x1c109a8) at explain.c:792
#15 0x00000000007eec54 in ExplainOnePlan (plannedstmt=0x1c0f9b0, into=0x0, es=0x7fff338e9360, 

```

**Comment by hlinnaka** [comment]
I believe this is a duplicate of https://github.com/greenplum-db/gpdb/issues/9404 or https://github.com/greenplum-db/gpdb/issues/9359. It works for me on fresh checkout of 6X_STABLE.

Please upgrade to the latest version, and try again. I'm closing this on the assumption that this has been fixed, but please re-open if you can reproduce this on a fresh checkout.

**Comment by chenbaggio** [comment]
so sorry ! it is not duplicate of #9404 or #9359. I have merged relative code and the problem is 
solved mentioned in the two issues, it is a new issue, please help to fix it, thanks!

**Comment by hlinnaka** [comment]
Ok, I can reproduce it now, I had assertions disabled earlier. I'll take a look, thanks for the report.

**Comment by jianlirong** [comment]
@hlinnaka ，when you investigate this issue, would you please also fix #9780 ? Thanks.

**Comment by hlinnaka** [comment]
> @hlinnaka ，when you investigate this issue, would you please also fix #9780 ? Thanks.

That seems unrelated to this.

**Comment by hlinnaka** [comment]
Pushed the fix to 6X_STABLE in commit 8be089d22c8a9a2b5fd563f0d228cb5291e502a8. Thanks for the report!
